### PR TITLE
push click tracking issue fix

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
Added fix for push click tracking not working issue
Improved logs for in-app and SDK initialisation

### CODE REVIEW

File: BlueShift-iOS-SDK/BlueShift.m [hunk 1]
```diff
@@ -172,14 +172,16 @@ - (void) setupWithConfiguration:(BlueShiftConfig *)config {
 
 /// Print debug logs on SDK initialization
 - (void)logSDKInitializationDetails {
-    [BlueshiftLog logInfo:@"SDK configured successfully. Below are the config details" withDetails:[_config getConfigStringToLog] methodName:nil];
-
-    if ([[BlueShiftAppData currentAppData] enablePush] == YES) {
-        [BlueshiftLog logInfo: @"EnablePush has been set to YES. If app push notification permission is accepted then app will start receiving push notifications from Blueshift." withDetails:nil methodName:nil];
-    } else {
-        [BlueshiftLog logInfo: @"EnablePush has been set to NO. The app will not receive any push notifications from Blueshift. To enable receiving push notifications, set enablePush to true and fire identify call from Blueshift SDK." withDetails:nil methodName:nil];
+    if (_sharedBlueShiftInstance.config.debug == YES) {
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShift.m:175]: do we do this check each time we log something in our SDK?

File: BlueShift-iOS-SDK/BlueShift.m [hunk 2]
```diff
@@ -172,14 +172,16 @@ - (void) setupWithConfiguration:(BlueShiftConfig *)config {
...
+        [BlueshiftLog logInfo:@"SDK configured successfully. Below are the config details" withDetails:[_config getConfigStringToLog] methodName:nil];
+        
+        if ([[BlueShiftAppData currentAppData] enablePush] == YES) {
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShift.m:178]: should we do this for enableInApp as well?

File: BlueShift-iOS-SDK/BlueShift.m (hunk 1 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShift.m:175]: `logSDKInitializationDetails` method logs the SDK configuration details at the time of SDK initialisation using `getConfigStringToLog` method call written at line 176. The method also prints few more logs for Push and enableTracking.

File: BlueShift-iOS-SDK/BlueShift.m (hunk 2 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShift.m:178]: Yes, I can add it for enableInApp.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**